lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -1060,6 +1086,80 @@ def _restart_opensearch(self, event: _RestartOpenSearch) -> None:
 
         self._start_opensearch_event.emit()
 
+    def _settle_voting_exclusions(self, unit_is_stopping: bool = False):  # noqa C901
+        """Settle the exclusions for all voting units."""
+        hosts = self.alt_hosts
+        if (
+            self.opensearch_peer_cm.deployment_desc().typ != DeploymentType.MAIN_ORCHESTRATOR
+            and (peer_cm_rel_data := self.opensearch_peer_cm.rel_data()) is not None
+        ):
+            # Also consider peer-relation units
+            hosts.extend([node.ip for node in peer_cm_rel_data.cm_nodes])
```

Comment:
This is not needed. `self.alt_hosts` is enough, it already contains all possible alternative hosts incl. from the large deployment relations.

**Note:** Please do not rely on `rel_data` for as long as you can - we should aim for a clear separation of concerns. The 2 checks can be replaced more robustly with `self.opensearch_peer_cm.is_consumer()`.

```suggestion
```

Reply:
If `self.alt_hosts` returns the right value in any case, do I still need that if and `self.opensearch_peer_cm.is_consumer()` then?